### ISSUE TITLE
Ticking block entity, autocrafting

### ISSUE THREAD

**Issue by JurJean** [raoulvdberge/refinedstorage#879]
#### Issue description:
The world became unusable due to crash on login. The server starts up fine. After login the attached crash report was generated.

#### What happens:
I provided a crafter with continuous redstone signal. I placed compressed cobblestone patterns in it. I turned on both redstone options on the crafter.

The server crashes with:
```
java.lang.NullPointerException: Ticking block entity
	at java.util.ArrayList.removeIf(ArrayList.java:1405)
	at com.raoulvdberge.refinedstorage.apiimpl.autocrafting.task.CraftingTask.update(CraftingTask.java:387)
	at com.raoulvdberge.refinedstorage.tile.TileController.func_73660_a(TileController.java:274)
	at net.minecraft.world.World.func_72939_s(World.java:1804)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:620)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:709)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:149)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471)
	at java.lang.Thread.run(Thread.java:745)
```

#### What you expected to happen:
All my cobblestone to be converted to compressed ones.

#### Steps to reproduce:
1. Put some patterns for (double,triple,..) compressed cobblestone into a crafter
2. Put a lever on it and turn it on
3. Insert massive amounts of cobblestone into the system

I don't recall to have turned on ore dictionary, so this was on the default setting. Also, it took a while for the server to crash. Just before, I noticed the crafting monitor to have 42341423423414 missing cobblestone (made up number, really high). Maybe ran out of int space? :P.

#### Version (Make sure you are on the latest version before reporting):

- Minecraft: 1.10.2
- Forge: 12.18.3.2185
- Refined Storage: 1.2.17

#### Does this issue occur on a server? [yes/no]
yes

#### If a (crash)log is relevant for this issue, link it here:

[crash-2017-01-17_13.05.43-server.txt](https://github.com/raoulvdberge/refinedstorage/files/710939/crash-2017-01-17_13.05.43-server.txt)

**Comment by way2muchnoise** [comment]
Compressing cobblestone with RS as an autocraft seems rather heavy on RS for no reason. You could use drawers or anything else to auto compress. RS isn't really made for doing that.

@raoulvdberge I think this is more of a autocraft of redstone issue than an auto crafting issue

**Comment by raoulvdberge** [comment]
I think this is fixed now?

**Comment by JurJean** [comment]
Not fixed, server still crashes. It doesn't matter if it's heavy, it should not crash.

**Comment by raoulvdberge** [comment]
Crashreport or it didn't happen.

**Comment by JurJean** [comment]
[crash-2017-03-07_19.25.36-server.txt](https://github.com/raoulvdberge/refinedstorage/files/825669/crash-2017-03-07_19.25.36-server.txt)
No problem!

**Comment by JurJean** [comment]
I tested the setup in a test world this weekend, and all seems to work fine. After getting the server working again, I noticed various invalidated patterns. Maybe this was the cause for the problems to persist.

I recreated the patterns and until now everything works awesome. I ran into another problem but I need to investigate more, I will post the results in another thread.

This issue can be closed. If anything happens I'll let you know.
